### ISSUE TITLE
Multiple Hosts + post-deploy

### ISSUE THREAD

**Issue by dnish** [andruschka/pm2-meteor#28]
Maybe we should add an option to deploy to multiple hosts? So we have to change the server object to an array and mix the environment variables (one global for all servers and one seperate for every single server).
## 

Adding post-deploy would execute a configurable command after successfully deploying the app, f.e. running an update command for <code>geoip-lite</code>.

**Comment by andruschka** [comment]
yes I am working on this one - but I have my own side business and working fulltime beneath.
It is really hard to find time  - would be nice if some would help me ^^

**Comment by dnish** [comment]
Think I'll try my best on my fork to add the mutli server deployment. If I got it right, it should be okay to modify https://github.com/andruschka/pm2-meteor/blob/master/lib/commandList.js#L29 

So we could read the server array and execute backup, ship etc. in a forEach loop while creating a copy of the pm2config object with the values for each single server (with modified environment variables).

**Comment by andruschka** [comment]
ah damn - you are modifying the js files - the lib folder gets rebuilt each deployment - I wrote pm2-meteor in coffeescript ...

**Comment by dnish** [comment]
Yeah I know, I'm not using Coffeescript but I think it shouldn't be a problem for me to write CS on another fork and then merge the changes. Just want to know if I could realize the multi server deployment on those lines or if I've forgotten something. The only problem I may see is that we have to built the app first and then do all other steps (f.e. check if the server has PM2 installed). This could be users for others who haven't configured their servers right.

**Comment by andruschka** [comment]
Yes that is right - this function will get called if you are using pm2-meteor deploy -
what should happen then is that a new remote task should check if the pm2-meteor.json file has multiple hosts - and if so - run the remote tasks for each machine.

**Comment by aleritty** [comment]
Please if you add an array of servers add also an "enabled" flag, so we can have someting like:

```
servers: [
  myserverOne:{
    enabled: true,
    host: "www.myserverone.com",
    pem: "myID.pem"
  },
  myserverTwo:{
    enabled:false,
    host:"www.IneedToRememberThisButNotDeploy.com",
    perm:"myID.pem"
  }
]
```

**Comment by satyavh** [comment]
+1 

I think this would totally make PM2-meteor a Mup alternative!

**Comment by andruschka** [comment]
Yep - I am working on a nee Version with some major changes as i wont give up Meteor- i was just really bussy the las few months.

**Comment by andruschka** [comment]
I also thought on something like pm2-meteor headup which will list all your apps with status details

**Comment by satyavh** [comment]
That would be super awesome!

**Comment by ryohkyo** [comment]
@andruschka not to be annoying or pushy, just want to check on the status regarding the multi-server deployment. :D

**Comment by Gamec** [comment]
Any progress on that one?

**Comment by deeeed** [comment]
+1

**Comment by flaviohenriquecbc** [comment]
+1

**Comment by adammoisa** [comment]
Still hoping for multiple host support... :)
